static/js/admin/src/AbstractForm.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Implements functionality and UI for the generic add container form
+ */
+define('AbstractForm', [ 'jquery', 'jquery-ui', 'underscore', 'RemoteStateChangeMonitor', 'ModalCreate',
```

Comment:
Since this is largely the same as AbstractFileUploadForm, it would seem to make sense for that to inherit from this.